### DISCUSSION TITLE
Search User by relation

### DISCUSSION THREAD

**Discussion by e12max** [OctopyID/LaraPersonate#9] [Ideas] [answered]
Hi! What do you think about to adding the ability to search for Users not only by self fields, but also relation fields?
For example:
in [ImpersonateController.php](https://github.com/OctopyID/LaraPersonate/blob/main/src/Http/Controllers/ImpersonateController.php) in function [list](https://github.com/OctopyID/LaraPersonate/blob/14da3ef815fceaabf4ad6845592dd89bf3e0b00c/src/Http/Controllers/ImpersonateController.php#L45) change [this](https://github.com/OctopyID/LaraPersonate/blob/14da3ef815fceaabf4ad6845592dd89bf3e0b00c/src/Http/Controllers/ImpersonateController.php#L62) lines:
```
foreach (config('impersonate.field.search_keys', []) as $field) {
    $query->orWhere($field, 'LIKE', '%' . $request->get('search') . '%');
}
```

on this:
```
foreach (config('impersonate.field.search_keys', []) as $field) {
    if (count($relation = explode('.', $field)) > 1) {
        $query->orWhereRelation($relation[0], $relation[1], 'LIKE', '%' . $request->get('search') . '%');
    } else {
        $query->orWhere($field, 'LIKE', '%' . $request->get('search') . '%');
    }
}

```

And in config file [search_keys](https://github.com/OctopyID/LaraPersonate/blob/main/config/impersonate.php#L70) can looks like
```
'search_keys' => [
    'name', 
    'email',
    'relation.field_name',
],
```

Or something like that) 
Thank you!

P.S. It is convenient to search for Users by roles, because almost all restrictions are associated with them.

**Comment by SupianIDz** [comment]
@e12max Very good idea, unfortunately I'm quite busy at the moment, can you open PR for this? I will merging them later. Thanks

**Comment by SupianIDz** [comment, marked as answer]
Added on [v3.0.0](https://github.com/OctopyID/LaraPersonate/releases/tag/v3.0.0)
